.github/workflows/build_documentation.yml

```diff
@@ -0,0 +1,88 @@
+name: Build documentation
+
+on:
+  push:
+    branches:
+      - main
+      - doc-builder*
+      - v*-release
+
+jobs:
+  build_and_package:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+
+    steps:
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '16'
+
+      - uses: actions/checkout@v2
+        with:
+          repository: 'huggingface/doc-builder'
+          path: doc-builder
+
+      - uses: actions/checkout@v2
+        with:
+          repository: 'huggingface/doc-build'
+          path: doc-build
+          token: ${{ secrets.HUGGINGFACE_PUSH }}
+
+      - uses: actions/checkout@v2
+        with:
+          repository: 'huggingface/optimum'
+          path: optimum
+
+      # - uses: actions/checkout@v2
+      #   with:
+      #     repository: 'huggingface/notebooks'
+      #     path: notebooks
+      #     token: ${{ secrets.HUGGINGFACE_PUSH }}
+
+      - name: Loading cache.
+        uses: actions/cache@v2
+        id: cache
+        with:
+          path: ~/.cache/pip
+          key: v1-test_build_doc
+          restore-keys: |
+            v1-test_build_doc-${{ hashFiles('setup.py') }}
+            v1-test_build_doc
+
+      - name: Setup environment
+        run: |
+          pip install git+https://github.com/huggingface/doc-builder
+          cd optimum
+          pip install .[dev]
+          cd ..
+
+      - name: Setup git
+        run: |
+          git config --global user.name "Hugging Face Doc Builder"
+          git config --global user.email docs@huggingface.co
+
+          cd doc-build
+          git pull origin main
+          cd ..
+
+      - name: Make documentation
+        run: |
+          version=$(pip show optimum | grep Version | cut -c 10-20) &&
+          cd doc-builder &&
+          doc-builder build optimum ../optimum/docs/source --build_dir ../doc-build --clean --version $version --html &&
```

Comment:
By default, `doc-builder` builds the `master` version of the docs when a `dev` substring is present in the `__version__` attribute. Here I extract the latest version number from the install and later we can think about how to accomodate the `main` / `master` build